web/packages/teleterm/src/ui/Vnet/VnetSliderStep.story.tsx

```diff
@@ -0,0 +1,169 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Box } from 'design';
+
+import { usePromiseRejectedOnUnmount } from 'shared/utils/wait';
+
+import { MockAppContextProvider } from 'teleterm/ui/fixtures/MockAppContextProvider';
+import { MockAppContext } from 'teleterm/ui/fixtures/mocks';
+import { MockedUnaryCall } from 'teleterm/services/tshd/cloneableClient';
+
+import { VnetContextProvider } from './vnetContext';
+import { VnetSliderStep } from './VnetSliderStep';
+
+export default {
+  title: 'Teleterm/Vnet/VnetSliderStep',
+  decorators: [
+    Story => {
+      return (
+        <Box width={324} bg="levels.elevated">
+          <Story />
+        </Box>
+      );
+    },
+  ],
+};
+
+const dnsZones = ['teleport.example.com', 'company.test'];
+
+export function Running() {
+  const appContext = new MockAppContext();
+  appContext.statePersistenceService.putState({
+    ...appContext.statePersistenceService.getState(),
+    vnet: { autoStart: true },
+  });
+  appContext.workspacesService.setState(draft => {
+    draft.isInitialized = true;
+  });
+  appContext.vnet.listDNSZones = () => new MockedUnaryCall({ dnsZones });
+
+  return (
+    <MockAppContextProvider appContext={appContext}>
+      <VnetContextProvider>
+        <Component />
+      </VnetContextProvider>
+    </MockAppContextProvider>
+  );
+}
+
+export function UpdatingDnsZones() {
+  const appContext = new MockAppContext();
+  appContext.statePersistenceService.putState({
+    ...appContext.statePersistenceService.getState(),
+    vnet: { autoStart: true },
+  });
+  appContext.workspacesService.setState(draft => {
+    draft.isInitialized = true;
+  });
+  const promise = usePromiseRejectedOnUnmount();
+  appContext.vnet.listDNSZones = async () => {
+    await promise;
+    return new MockedUnaryCall({ dnsZones: [] });
+  };
+
+  return (
+    <MockAppContextProvider appContext={appContext}>
+      <VnetContextProvider>
+        <Component />
+      </VnetContextProvider>
+    </MockAppContextProvider>
+  );
+}
+
+export function DnsZonesError() {
```

Comment:
This story crashes:
<img width="740" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/fd6814b8-ed49-4fb3-9c93-883b8b3c73dc">

Reply:
Thanks for the catch. This is because the "mocked" `listDNSZones` was not an async function and `MockedUnaryCall` did not implement `then` properly.

```
  appContext.vnet.listDNSZones = () =>
    new MockedUnaryCall(undefined, new Error('something went wrong'));
```